smoke-tests/apps/RoleNameOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/AksRoleNameOverridesTest.java

```diff
@@ -35,7 +35,13 @@ abstract class AksRoleNameOverridesTest {
   @Test
   @TargetUri("/testAks")
   void test() throws Exception {
-    verifyRoleNameAndInstance("test-deployment-name", "test-pod-name");
+    String jbossHome = System.getenv("JBOSS_HOME");
+    if (jbossHome != null) {
+      // jboss/wildfly has a default service.name="jboss-module"
```

Comment:
😮

Reply:
upstream added it correct? it didn't have it before though

Reply:
oh, I understand. Upstream has some service name detectors, and now with latest azure-sdk changes, we're factoring in otel resource `service.name` into role name. this could be breaking change for Java agent users. I think we should override exactly which (if any) resource providers (detectors) we want to include in our Java agent distro

Reply:
see https://github.com/open-telemetry/opentelemetry-java/blob/e06d35add577fd702104b5b2836201e76f72d88b/sdk-extensions/autoconfigure/README.md?plain=1#L203

Reply:
we may need just one resource provider: `io.opentelemetry.sdk.autoconfigure.internal.EnvironmentResourceProvider`